src/lib/encoder.ts

```diff
@@ -313,20 +315,32 @@ const decodeStringCompactBytesArray = (compactBytesArray: string): string[] => {
 
 const valueTypeEncodingMap = {
   bool: {
-    encode: (value: boolean) => abiCoder.encodeParameter('bool', value),
-    decode: (value: string) => abiCoder.decodeParameter('bool', value),
+    encode: (value: boolean) => (value ? '0x01' : '0x00'),
+    decode: (value: string) => value === '0x01',
   },
   boolean: {
-    encode: (value: boolean) => abiCoder.encodeParameter('bool', value),
-    decode: (value: string) => abiCoder.decodeParameter('bool', value),
+    encode: (value: boolean) => (value ? '0x01' : '0x00'),
+    decode: (value: string) => value === '0x01',
   },
   string: {
-    encode: (value: string) => abiCoder.encodeParameter('string', value),
-    decode: (value: string) => abiCoder.decodeParameter('string', value),
+    encode: (value: string | number) => {
+      // if we receive a number as input,
+      // convert each letter to its utf8 hex representation
+      if (typeof value === 'number') {
+        return utf8ToHex(`${value}`);
+      }
+
+      return utf8ToHex(value);
+    },
+    decode: (value: string) => hexToUtf8(value),
   },
   address: {
-    encode: (value: string) => abiCoder.encodeParameter('address', value),
-    decode: (value: string) => abiCoder.decodeParameter('address', value),
+    encode: (value: string) => {
+      const abiEncodedValue = abiCoder.encodeParameter('address', value);
+      const bytesArray = hexToBytes(abiEncodedValue);
+      return bytesToHex(bytesArray.slice(12));
+    },
+    decode: (value: string) => toChecksumAddress(value),
```

Comment:
We had issues with the lib, like if someone writes bad data in the contract, when we decode, it breaks.

Ideally, we want our system to not crash and simply return nothing or skip the value, or maybe flag it as invalid. Without breaking the whole thing.

I think we should make a big check and make sure that errors are properly cought so it doesnt break a "decode all data" function when on of them is corrupted.